develop/toolkit/api-libraries/smart-contracts/ethers-js.md

```diff
@@ -0,0 +1,113 @@
+
+## Set Up the Project
+
+To start working with Ethers.js, begin by initializing your project:
+
+```bash
+npm init -y
+```
+
+## Install Dependencies
+
+Next, install the Ethers.js library:
+
+```bash
+npm install ethers
+```
+
+## Set Up the Ethers.js Provider
+
+To interact with the Asset Hub, you'll need to set up an Ethers.js provider. This provider connects to a blockchain node, allowing you to query blockchain data and interact with smart contracts. Here's how to configure it:
+
+```js
+--8<-- 'code/develop/smart-contracts/evm-toolkit/ethers-js/connectToProvider.js'
+```
+
+!!! note
+    Replace `INSERT_RPC_URL`, `INSERT_CHAIN_ID`, and `INSERT_CHAIN_NAME` with the appropriate values. For example, to connect to Westend Asset Hub's Ethereum RPC instance, you can use the following parameters:
+
+    ```js
+    const PROVIDER_RPC = {
+        rpc: 'https://westend-asset-hub-eth-rpc.polkadot.io',
+        chainId: 420420421,
+        name: 'westend-asset-hub'
+    };
+    ```
+
+With the [`Provider`](https://docs.ethers.org/v6/api/providers/#Provider){target=\_blank} set up, you can start querying the blockchain. For instance, to fetch the latest block number:
+
+??? code "Fetch Last Block code"
+
+    ```js
+    --8<-- 'code/develop/smart-contracts/evm-toolkit/ethers-js/fetchLastBlock.js'
+    ```
+
+## Compile Contracts
+
+To deploy smart contracts to Asset Hub, you need to compile them into `polkavm` bytecode. Use the [`@parity/revive`](https://www.npmjs.com/package/@parity/revive){target=\_blank} library, which compiles Solidity code for use on Substrate-based chains.
+
+Install the `@parity/revive` library:
+
+```bash
+npm install --save-dev @parity/revive 
+```
+### Example: Storage.sol
+
+Here's a sample Solidity contract (`Storage.sol`) to be compiled and deployed to the Asset Hub. This contract's functionality stores a number and permits users to update it with a new value.
+
+??? code "Storage.sol"
+
+    ```solidity
+    --8<-- 'code/develop/smart-contracts/evm-toolkit/ethers-js/Storage.sol'
+    ```
+
+To compile this contract, use the following script:
+
+```js
+--8<-- 'code/develop/smart-contracts/evm-toolkit/ethers-js/compile.js'
+```
+
+Note that the script above is tailored to the `Storage.sol` contract. It can be adjusted for other contracts by changing the file name or modifying the ABI and bytecode paths accordingly.
+
+After executing the script, the Solidity contract will be compiled into the required `polkavm` bytecode format. The ABI and bytecode will be saved into files with `.json` and `.polkavm` extensions, respectively. You can now proceed with deploying the contract to the Asset Hub network, as outlined in the next section.
+
+## Contracts Deployment
+
+To deploy the compiled contract to the Asset Hub, you will need a wallet with a private key to sign the deployment transaction. You can use [subkey](/polkadot-protocol/basics/accounts/#using-subkey){target=\_blank} to manage your wallet.
```

Comment:
I don't think subkey is the best example here, as you could use any other tool that's being used in evm development